java/src/com/ibm/streamsx/rest/StreamingAnalyticsConnection.java

```
@@ -29,50 +42,91 @@
             JsonObject resources = jParse.parse(sResources).getAsJsonObject();
 
             restURL = resources.get("streams_rest_url").getAsString();
         }
 
-        if ( restURL.equals("") ) {
+        if (restURL.equals("")) {
             throw new IllegalStateException("Missing restURL for service");
         }
 
-        SAConn.setURL( restURL ) ;
+        streamingConnection.setURL(restURL);
         String[] rTokens = resourcesPath.split("/");
         if (rTokens[3].equals("service_instances")) {
-            SAConn.setInstanceId(rTokens[4]) ;
+            streamingConnection.setInstanceId(rTokens[4]);
         } else {
             throw new IllegalStateException("Resource Path decoding error.");
         }
-        return SAConn ;
+        return streamingConnection;
+    }
+
+    private void setJobsPath(String sJobs) {
+        jobsPath = sJobs;
+    }
+
+    /**
+     * @param jobId
+     *            string indicating the job id to be cancelled
+     * @return true if job is cancelled, false otherwise
+     */
+    @Override
+    public boolean cancel_job(String jobId) {
+        boolean rc = false;
+        String sReturn = "";
+        try {
+            String deleteJob = jobsPath + "?job_id=" + jobId;
+
+            Request request = Request.Delete(deleteJob).addHeader(AUTH.WWW_AUTH_RESP, apiKey).useExpectContinue();
+
+            Response response = executor.execute(request);
+            HttpResponse hResponse = response.returnResponse();
+            int rcResponse = hResponse.getStatusLine().getStatusCode();
+
+            if (HttpStatus.SC_OK == rcResponse) {
+                sReturn = EntityUtils.toString(hResponse.getEntity());
+                rc = true;
+            } else {
+                rc = false;
+            }
+            traceLog.finest("Request: [" + deleteJob + "]");
+            traceLog.finest(rcResponse + ": " + sReturn);
+        } catch (IOException e) {
+            rc = false;
```

Don't hide exceptions, it causes immense pain for support.